### PR TITLE
run e2e test only against kind

### PR DESCRIPTION
**Pull request by sthaha**

### CODE REVIEW

File: hack/cluster.sh [hunk 1]
```diff
@@ -90,14 +96,24 @@ main() {
 	local op="$1"
 	shift
 	cd "$PROJECT_ROOT"
-
-	# NOTE: all operations are relative to tmp
+	export PATH="$BIN_DIR:$PATH"
 	mkdir -p "${TMP_DIR}"
-	git_checkout
+	ensure_all_tools
```
Comment (vprashar2929) [inline hack/cluster.sh:101]: Do we need to call ensure_all_tools at every operation? I think we should call this during cluster_prereqs step or during on_cluster_up. Wdyt?

File: hack/cluster.sh [hunk 2]
```diff
@@ -90,14 +96,24 @@ main() {
...
+
+	#NOTE: allow cluster_<OP> to executed without going through the entire
+	# setup again. This is useful (especially) in CI which uses the kepler-action
+	# to setup a cluster which does not have all the prerequisites for running
+	# Operator as OLM Bundle
+	declare -F | cut -f3 -d ' ' | grep -qE "^cluster_$op\$" && {
+		header "Running cluster_$op"
+		"cluster_$op" "$@"
```
Comment (vprashar2929) [inline hack/cluster.sh:109]: Is there any need to pass args to cluster_prereqs?

File: tests/run-e2e.sh [hunk 3]
```diff
@@ -18,8 +25,155 @@ declare SHOW_USAGE=false
...
+			grep -i error | tee "$error_log"
+		line 50
+	fi
+
+	return $ret
+}
+
+parse_args() {
+	### while there are args parse them
+	while [[ -n "${1+xxx}" ]]; do
```
Comment (vprashar2929) [inline tests/run-e2e.sh:140]: In case of no args passed while running `run-e2e.sh` shouldn't we show help and exit?

File: hack/cluster.sh (hunk 2 above)
Comment (sthaha) [reply hack/cluster.sh:109]: this applies to all `cluster_op` so it is best to pass all args to that function .

File: tests/run-e2e.sh (hunk 3 above)
Comment (sthaha) [reply tests/run-e2e.sh:140]: no, `./test/run-e2e.sh` is a valid invocation.

File: hack/cluster.sh (hunk 1 above)
Comment (sthaha) [reply hack/cluster.sh:101]: yes, always best to ensure all tools are present. This is an assumption that all functions can make. thus no need to check if `x, y, z` is installed.

Comment (vprashar2929) [reply hack/cluster.sh:101]: Why need this in case of cluster_down?

Comment (sthaha) [reply hack/cluster.sh:101]: The same applies,  ... it is best to ensure all tools exist before starting anything and for a non - production script, these things do not matter.

File: hack/cluster.sh (hunk 2 above)
Comment (vprashar2929) [reply hack/cluster.sh:109]: We only have cluster_prereqs and we don't use args there. What's the point of passing args there?

**Comment by vprashar2929** [comment]
Also please make change [here](https://github.com/sustainable-computing-io/kepler-operator/blob/c6ee1f49e837221c506bb2e649f155a2c9fcc6b6/tests/run-e2e.sh#L192) as well😅

File: tests/run-e2e.sh [hunk 4]
```diff
@@ -18,8 +25,155 @@ declare SHOW_USAGE=false
...
+	kubectl get events -w \
+		-o custom-columns=FirstSeen:.firstTimestamp,LastSeen:.lastTimestamp,Count:.count,From:.source.component,Type:.type,Reason:.reason,Message:.message \
+		-n "$ns" | tee "$LOGS_DIR/$ns-events.log"
+}
+
+watch_operator_errors() {
+	local err_log="$1"
+	shift
+
+	kubectl logs -f -n "$OPERATORS_NS" "deploy/$OPERATOR_DEPLOY_NAME" |
```
Comment (vprashar2929) [inline tests/run-e2e.sh:102]: We shouldn't use -f here.

**Comment by vprashar2929** [comment]
We can fix some of the issues within the script in later PR but the initial implementation works fine

**Review by vprashar2929 — APPROVED** [review]
LGTM

File: tests/run-e2e.sh (hunk 4 above)
Comment (sthaha) [reply tests/run-e2e.sh:102]: we should use -f here :) .. it looks only for errors. See the invocation below where this is run in background.

File: tests/run-e2e.sh (hunk 3 above)
Comment (sthaha) [reply tests/run-e2e.sh:140]: When you run locally which does a complete build and deploy of the operator and bundle.

**Pull request merged**